velox/common/encode/Base32.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/Range.h>
+#include <folly/io/IOBuf.h>
+
+#include <array>
+#include <string>
+
+#include "velox/common/base/GTestMacros.h"
+#include "velox/common/base/Status.h"
+
+namespace facebook::velox::encoding {
+
+class Base32 {
+ public:
+  static const size_t kCharsetSize = 32;
+  static const size_t kReverseIndexSize = 256;
+
+  /// Character set used for Base32 encoding.
+  /// Contains specific characters that form the encoding scheme.
+  using Charset = std::array<char, kCharsetSize>;
+
+  /// Reverse lookup table for decoding.
+  /// Maps each possible encoded character to its corresponding numeric value
+  /// within the encoding base.
+  using ReverseIndex = std::array<uint8_t, kReverseIndexSize>;
+
+  /// Encodes the specified number of characters from the 'input'.
+  static std::string encode(const char* input, size_t inputSize);
+
+  /// Encodes the specified text.
+  static std::string encode(folly::StringPiece text);
+
+  /// Encodes the specified IOBuf data.
+  static std::string encode(const folly::IOBuf* inputBuffer);
+
+  /// Encodes the specified number of characters from the 'input' and writes the
+  /// result to the 'outputBuffer'. The output must have enough space as
+  /// returned by the calculateEncodedSize().
+  static void encode(const char* input, size_t inputSize, char* outputBuffer);
```

Comment:
Shouldnt it return the size of bytes written to outputBuffer ? (or perhaps a stringview ?)

Reply:
Updated to return size_t